README.md

```diff
@@ -1,20 +1,81 @@
-# Project/Repo Title
+# Docker Build Action
 
-Template Repository for the Boutros Lab general project repos. Describe a simple overview of use/purpose here.
+An Action to automatically build and push images to the [GitHub Container registry](https://github.com/features/packages).
 
 ## Description
 
-An in-depth paragraph about your project and overview of use.
+This action will build and push images of the form `ghcr.io/<organization>/<image>:<version>`. The `<version>` field is controlled by the following logic:
 
-## License
+| Pushed Ref | Type | Resulting Tag |
+| ---------- | -------------- | ----------------- |
+| `main` | default branch | `dev` |
+| `mybranch` | branch | `branch-mybranch` |
+| `v1.2.3` | tag | `1.2.3` |
+When a git branch or tag is deleted, the corresponding docker will be deleted as well.
+
+## Usage
+
+```yaml
+---
+name: Update image in GHCR
+
+run-name: >
+  ${{
+    github.event_name == 'delete' && format(
+      'Delete `{0}{1}`',
+      github.event.ref_type == 'branch' && 'branch-' || '',
+      github.event.ref
+    )
+    || github.ref == 'refs/heads/main' && 'Update `dev`'
+      || format(
+        'Update `{0}{1}`',
+        !startsWith(github.ref, 'refs/tags') && 'branch-' || '',
+        github.ref_name
+      )
+  }} docker tag
+
+on:
+  push:
+    branches-ignore: ['gh-pages']
+    tags: ['v*']
+  delete:
+
+jobs:
+  push-or-delete-image:
+    runs-on: ubuntu-latest
+    name: Update GitHub Container Registry
+    permissions:
+      contents: read
+      packages: write
+    steps:
+      - uses: uclahs-cds/tool-Docker-action@v2
+```
 
-Author: Name1(username1@mednet.ucla.edu), Name2(username2@mednet.ucla.edu)
+The complicated `run-name` logic above controls the workflow run names listed on the Actions page:
+
+| Ref Name | Ref Type | `push` Run Name | `delete` Run Name |
+| -------------------- | -------- | ----------------------------------- | ----------------------------------- |
+| Push to `main` | branch | Update `dev` docker tag | Delete `dev` docker tag |
+| Push to `mybranch` | branch | Update `branch-mybranch` docker tag | Delete `branch-mybranch` docker tag |
+| Push to `v1.2.3` tag | tag | Update `v1.2.3` docker tag | Delete `v1.2.3` docker tag |
```

Comment:
```suggestion
| Push to `v1.2.3` tag | tag | Update `1.2.3` docker tag | Delete `1.2.3` docker tag |
```

The Docker tags don't include the `v` I believe

Reply:
_Correct_, the description incorrectly includes the leading `v`, but I can't fix that (see the Errata above). I thought this slightly-incorrect string was more useful than "Update tag", but I'm open to feedback on that.

Reply:
Ah got it, this table is indicating the actual run name(s) that would get generated

Reply:
Correct - I've got the actual tag names (showing that `v`-loss between git and docker) in the prior table.